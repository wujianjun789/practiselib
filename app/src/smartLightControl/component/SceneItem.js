/**
 * Created by a on 2017/9/11.
 */
import React,{Component} from 'react'

export default class SceneItem extends Component{
    constructor(props){
        super(props);

        this.activeClick = this.activeClick.bind(this);
    }

    activeClick(){
        const {id, activeClick}  = this.props;
        activeClick && activeClick(id);
    }

    render(){
        const {className, name, active, asset} = this.props;

        let assetName = "";
        asset.map((ass,index)=>{
            assetName += ass.name+(index<asset.length-1?"，":"");
        })
       return <div className={"panel panel-default scene-item "+className}>
           <div className="panel-heading">
               {name}
               <button className="btn btn-primary" onClick={this.activeClick}>激活</button>
           </div>
           <div className="panel-body">
               <span>包含设备:&nbsp;&nbsp;</span>{assetName}
           </div>
        </div>
    }
}