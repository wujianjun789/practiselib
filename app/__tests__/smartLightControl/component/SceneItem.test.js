/**
 * Created by a on 2017/9/29.
 */
import React from 'react';
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer';

import SceneItem from '../../../src/smartLightControl/component/SceneItem';

test('renders SceneItem', ()=>{
    const sceneItem = renderer.create(
        <SceneItem />
    ).toJSON();

    expect(sceneItem).toMatchSnapshot();
});

test('renders SceneItem class', ()=>{
    const component = shallow(<SceneItem />)
    expect(component.find('.scene-item').length).toEqual(1);
})