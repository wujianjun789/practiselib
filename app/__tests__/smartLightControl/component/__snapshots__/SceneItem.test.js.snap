exports[`test renders SceneItem 1`] = `
<div
  className="panel panel-default scene-item undefined">
  <div
    className="panel-heading">
    <button
      className="btn btn-primary"
      onClick={[Function]}>
      激活
    </button>
  </div>
  <div
    className="panel-body">
    <span>
      包含设备:  
    </span>
    
  </div>
</div>
`;
